docs/nidigital/class.rst

```diff
@@ -1243,60 +1243,51 @@ frequency_counter_measure_frequency
 
 
 
-get_channel_name
-----------------
+get_channel_names
+-----------------
 
     .. py:currentmodule:: nidigital.Session
 
-    .. py:method:: get_channel_name(index)
+    .. py:method:: get_channel_names(indices)
 
-            TBD
+            Returns a list of channel names for given channel indices.
 
             
 
 
 
-            :param index:
-
-
-                
+            :param indices:
 
 
-            :type index: int
-
-            :rtype: str
-            :return:
-
-
-                    
-
-
-
-get_channel_name_from_string
-----------------------------
-
-    .. py:currentmodule:: nidigital.Session
-
-    .. py:method:: get_channel_name_from_string(index)
-
-            TBD
-
-            
-
-
-
-            :param index:
+                Specifies indices for the channels in the session.
+                Valid values are from zero to the total number of channels in the session minus one.
+                The following types and formats are supported:
+                  - Basic Sequence types
+                    - list - for example, [0, 2, 3, 1]
+                    - tuple - for example, (0, 2, 3, 1)
+                    - range - for example, range(0, 15, 2)
+                  - slice - for example, slice(0, 15, 2)
+                  - str
+                    - A comma-separated list - for example, "0,2,3,1"
+                    - A range using a hyphen - for example, "0-3"
+                    - A range using a colon - for example, "0:3"
+                  - int - for example, 0
 
+                The input can contain any combination of above types. Both out-of-order and repeated indices are
+                supported ("2,3,0", "1,2,2,3"). White space characters, including spaces, tabs, feeds, and
+                carriage returns, are allowed within strings. Ranges can be incrementing or decrementing.
 
                 
 
 
-            :type index: str
+            :type indices: basic sequence types or str or int
 
-            :rtype: str
+            :rtype: list of str
```

Comment:
I think most clients are going to call this with a single index: `my_session.get_channel_names(0)`. It's going to be annoying that it returns a list.

I'm not sure if that's ok or not.

Reply:
Neither is perfect, but we decided offline to use `list`.

Reply:
Let's leave it as list which is the most semantically accurate thing. I hope we don't regret it ever :)
I think in most cases the results are passed to the `channels` repeated capability container as-is so nobody will care to even inspect them.